.github/workflows/go.yml

```diff
@@ -49,7 +49,6 @@ jobs:
   go-test:
     needs: meta
     if: needs.meta.outputs.changed == 'true'
```

Comment:
It would still be valuable to have a timeout on the workflow itself in case we hang in another step. Maybe 30 mins?